src/main/java/net/fabricmc/loader/api/metadata/ModDependency.java

```diff
@@ -42,5 +49,51 @@
 	 *
 	 * @return representation of the dependency's version requirements
 	 */
-	Set<VersionPredicate> getVersionRequirements();
+	Collection<VersionPredicate> getVersionRequirements();
+
+	enum Kind {
```

Comment:
"type" is too overloaded with other meanings in this concept, relation* is wrong since ModDependency itself is the relation. Not sure what's wrong with calling depends, recommends , ... kinds of dependencies.